packages/js-client/GUIDE.md

```diff
@@ -0,0 +1,445 @@
+
+Aragon common packages will handle the connection to IPFS, Subgraph and blockchain providers, so you don't have to reinvent the wheel.
+
+Additionally, is giving some structure to your plugin, if you plan on doing some complex logic in your plugin, it is a good idea to have a JS client to handle it and leave the UI handling only the UI logic.
+
+## Getting started
+
+To get started with your client the first step is to create a nes JS/TS project you can use any tool you want.
+
+Once you have your project set up yo need to install the `@aragon/sdk-client-common` package.
+
+```bash
+npm install @aragon/sdk-client-common
+// or
+yarn add @aragon/sdk-client-common
+```
+
+Once you have the package installed you can start building your client.
+
+There are 3 main files that you will need to build your client:
+
+- `src/client.ts`
+- `src/context.ts`
+- `src/types.ts`
+
+### `src/types.ts`
+
+This file will include the types that your client will export, the first type that you will need will be the `ContextParams` type, this type will be used to define the parameters that your client will receive and it should be an extension of the ContextParams type from `@aragon/sdk-client-common`. This params will be later stored in the context and accessible by the modules, so if you have some endpoint, address or any other value that you need to access from your modules you should add it here.
+
+There are two additional types that you will need to add in this file.
+
+- `MyPluginState`: This type will be used to define the state of your client, this state will be stored in the context and it will be accessible by the modules. This type should extend the `ClientState` type from `@aragon/sdk-client-common`.
+
+- `MyPluginOverriddenState`: This type will be used to track what values have been modified and what values are default. This type should also extend the `OverriddenState` type from `@aragon/sdk-client-common`.
+
+In this file you should also define the state type of your client, this should extend the `ClientState` type from `@aragon/sdk-client-common`. This type will be used to define the state of your client, this state will be stored in the context and it will be accessible by the modules.
+
+```ts
+// src/types.ts
+import {ContextParams} from '@aragon/sdk-client-common';
+
+export type MyPluginClientParams = ContextParams & {
+  myCustomParam: string;
+  myCustomBackendUrl: string;
+  // Your custom params
+};
+
+export type MyPluginState = ClientState & {
+  myCustomParam: string;
+  // In the context you will be able to work with the params
+  // so the state can be a client for a service instead of the
+  // actual value
+  myCustomBackendClient: MyCustomBackendClient[];
+};
+
+export type MyPluginOverriddenState = OverriddenState & {
+  [key in keyof MyPluginContextState]: boolean;
+};
```

Comment:
Im a little lost here. There isnt a good explanation of what these types are and when you look in the template its not obvious as to what they map to. especially the back end client

Reply:
I added some more explaining lines, please re review it

Reply:
Awesome!
I have not followed the the guide and built a client yet but reading through it makes sense and im no longer lost  =]